thriftbp/server_middlewares.go

```diff
@@ -219,8 +219,15 @@ func ExtractDeadlineBudget(name string, next thrift.TProcessorFunction) thrift.T
 				v, err := strconv.ParseInt(s, 10, 64)
 				if err == nil && v >= 1 {
 					var cancel context.CancelFunc
```

Comment:
keep this right above the `ctx, cancel` line since it's being added so that line can avoid shadowing ctx

bonus points, `// avoid shadowing ctx`